contracts/axelarnet-gateway/src/contract.rs

```diff
@@ -76,7 +77,7 @@ pub fn execute(
         } => execute::call_contract(
             deps.storage,
             deps.querier,
-            info.sender,
+            info,
```

Comment:
Create a separate `ContractCalledWithToken` msg type. The routing is quite different between the two call types and call contract with token has more restrictions (token can only be a gateway token, and destination chain has to be a consensus connection), so I prefer to not combine them. This also makes it consistent with the legacy EVM gateway API of having two entrypoints.

Reply:
It feels awkward to create a separate `CallContractWithToken` where its msg struct is identical to `CallContract`, i.e. `token` is no where seen in its msg struct.

Reply:
Yeah, I understand, but I still find it worse to allow `CallContract` to support it given all the asterisks around legacy token/legacy chains, and since it'll be eventually removed. Would like to hear thoughts from others as well.

Reply:
I don't see why is it worse but we can see what others think.

Reply:
Oh ya, that as well. We didn't have this option for the EVM gateways since ERC20 tokens are not first-class citizens. In Cosmwasm, however, things are different.